src/basic/env-util.c

```diff
@@ -1167,7 +1167,7 @@ int setenvf(const char *name, bool overwrite, const char *valuef, ...) {
 
         /* Try to suppress writes if the value is already set correctly (simply because memory management of
          * environment variables sucks a bit. */
-        if (streq_ptr(getenv(name), value))
```

Comment:
This shouldn't be switched over. We're a setter anyways, so we expect what we're setting can be read back.